analytics/build/build.go

```diff
@@ -130,3 +133,37 @@ func evaluateActivities(rw *openrtb_ext.RequestWrapper, ac privacy.ActivityContr
 	cloneReq.RebuildRequest()
 	return true, cloneReq
 }
+
+func updateReqWrapperForAnalytics(rw *openrtb_ext.RequestWrapper, adapterName string) {
+	reqExt, _ := rw.GetRequestExt() // TODO: Handle error
+	reqExtPrebid := reqExt.GetPrebid()
+	if reqExtPrebid == nil {
+		return
+	}
+	extPrebidAnalytics := reqExtPrebid.Analytics
+
+	// Given analytics adapter module is not present in ext.prebid.analytics, remove the entire analytics object from request for logging
+	if _, ok := extPrebidAnalytics[adapterName]; !ok {
+		reqExtPrebid.Analytics = nil
+		reqExt.SetPrebid(reqExtPrebid)
```

Comment:
At this point, we will be modifying the original request wrapper sent in to `LogAuctionObject()` at the top of this file unless `evaluateActivities()` found something it needed to change and thus has already cloned the request wrapper. You probably need to record a `isCloned` flag and pass it down here, so you can know if you need to clone the request wrapper before modifying the Analytics object.

Reply:
My latest commits should address this, and I added tests that should've caught this issue initially!